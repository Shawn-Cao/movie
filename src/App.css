.App {
  text-align: center;
}

.heading-movies {
  color: red;
  text-align: left;
  animation: showup 2s;
  position: relative;
  margin-bottom: -20px;
  margin-left: 60px;
}

@keyframes showup {
    from { top: 20px; }
    to   { top: 0; }
}

.App-intro {
  font-size: large;
}

.box {
    width: 100px;
    height: 100px;
    -webkit-transition: width 1s, height 1s;
    transition: width 1s, height 1s;
}

.box:hover {
    width: 200px;
    height: 200px;
}
