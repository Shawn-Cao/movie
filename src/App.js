import React, { Component } from 'react';
import './App.css';
import Movies from './Movie';

class App extends Component {
  render() { //TODO: clear up here
    return (
      <div className="App">
        <h1 className="heading-movies">{'POPULAR MOVIES'}</h1>
        <Movies width={850} height={600}/>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;
