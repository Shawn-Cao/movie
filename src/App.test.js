jest.mock('./Movie/api/theMovieDb');
var dummyElement = global.document.createElement('div');
global.document.getElementById = jasmine.createSpy('HTML Element').and.returnValue(dummyElement);

import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
// });

describe('App', () => {
  it('renders without crashing and matching snapshot', () =>{
    const tree = renderer.create(
      <App />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
