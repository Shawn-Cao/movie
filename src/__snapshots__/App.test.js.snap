// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing and matching snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="heading-movies"
  >
    POPULAR MOVIES
  </h1>
  <div
    style={
      Object {
        "height": 600,
        "position": "relative",
        "width": 850,
      }
    }
  >
    <div
      className={undefined}
      style={
        Object {
          "left": "50%",
          "position": "absolute",
          "top": "50%",
          "zIndex": 1000,
        }
      }
    >
      <img
        alt="loading spinner"
        className="Spinner"
        src="logo.svg"
      />
    </div>
    <div
      id="movie-widget"
      style={
        Object {
          "height": 600,
          "overflowY": "scroll",
          "position": "relative",
          "width": 850,
        }
      }
    >
      <ul
        className="Movie"
      />
    </div>
  </div>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;
